### PR TITLE
Bug 1699411 - Reduce filled bug summary

### PR DESCRIPTION
**Pull request by beatrice-acasandrei**

### CODE REVIEW

File: ui/perfherder/helpers.js [hunk 1]
```diff
@@ -527,28 +529,63 @@ export const getTitle = (alertSummary) => {
...
+    ...new Set(
+      alertsInSummary.map((alert) => getTestName(alert.series_signature)),
+    ),
+  ];
+  let testInfo = tests.sort().join(' / ');
+  console.log('test ', testInfo);
+  console.log('max ', maxMagnitudeAlert);
+  console.log('min ', minMagnitudeAlert);
+  if (maxMagnitudeAlert && minMagnitudeAlert) {
+    testInfo = `${getTestName(
```
Comment (ionutgoldan) [inline ui/perfherder/helpers.js:562]: Found a bug. For a single regression, this will duplicate its name

**Review by ionutgoldan — CHANGES_REQUESTED** [review]

File: ui/perfherder/helpers.js [hunk 2]
```diff
@@ -527,28 +529,63 @@ export const getTitle = (alertSummary) => {
...
-  ]
-    .sort()
-    .join(' / ');
+  const tests = [
+    ...new Set(
+      alertsInSummary.map((alert) => getTestName(alert.series_signature)),
+    ),
+  ];
+  let testInfo = tests.sort().join(' / ');
+  console.log('test ', testInfo);
```
Comment (ionutgoldan) [inline ui/perfherder/helpers.js:558]: Be sure to remove these console logs.

File: ui/perfherder/helpers.js [hunk 3]
```diff
@@ -527,28 +529,63 @@ export const getTitle = (alertSummary) => {
...
+  const platforms = [
+    ...new Set(
+      alertsInSummary.map((alert) => alert.series_signature.machine_platform),
+    ),
+  ];
+  const availablePlatforms = ['Windows', 'Linux', 'OSX', 'Android'];
+
+  let platformInfo = [];
+  platforms.forEach((platform) =>
+    availablePlatforms.forEach((name) => {
```
Comment (ionutgoldan) [inline ui/perfherder/helpers.js:580]: This is a bit hard to read. It'd be nice to extract the generic name lookup into a separate method.

File: ui/perfherder/helpers.js [hunk 4]
```diff
@@ -527,28 +529,63 @@ export const getTitle = (alertSummary) => {
...
-    ...new Set(alertsInSummary.map((a) => a.series_signature.machine_platform)),
-  ]
-    .sort()
-    .join(', ');
+  const platforms = [
+    ...new Set(
+      alertsInSummary.map((alert) => alert.series_signature.machine_platform),
+    ),
+  ];
+  const availablePlatforms = ['Windows', 'Linux', 'OSX', 'Android'];
```
Comment (ionutgoldan) [inline ui/perfherder/helpers.js:576]: Could we move this into the `constants` module?

**Review by ionutgoldan — CHANGES_REQUESTED** [review]

File: ui/perfherder/helpers.js (hunk 1 above)
Comment (beatrice-acasandrei) [reply ui/perfherder/helpers.js:574]: For a single regression `maxMagnitudeAlert` and `minMagnitudeAlert` will have the value of undefined, line 546 which contains condition ` if (alertsInSummary.length > 1)` will make sure of this.

Comment (ionutgoldan) [reply ui/perfherder/helpers.js:574]: The bug will still reproduce if you have a summary with one regression and multiple improvements.

**Review by ionutgoldan — CHANGES_REQUESTED** [review]

**Comment by ionutgoldan** [comment]
Found a new bug.

#### Steps to reproduce
1. Go to an alert summary with multiple alerts (some of which were reassigned). ([example](http://localhost:5000/perfherder/alerts?id=29341&hideDwnToInv=0))
2. From the top right dropdown, click on **File bug**.
3. Click the **File Bug** button from the dialog.
4. Wait to be redirected to Bugzilla autofilled form.

Expected result:
Summary mentions there are "+ 11 more" perf regressions.

Actual result:
Summary mentions there are "+ 17 more" perf regressions.

**Comment by ionutgoldan** [comment]
@beatrice-acasandrei I also noticed this PR changes all summary headers from Alerts view. We shouldn't do this at all.

**Review by ionutgoldan — CHANGES_REQUESTED** [review]
We seem to count the invalid alerts, according to the summary. We should ignore those as if they even exist.

File: ui/perfherder/alerts/StatusDropdown.jsx [hunk 5]
```diff
@@ -124,9 +124,9 @@ export default class StatusDropdown extends React.Component {
       'ddd MMMM D YYYY',
     );
 
-    const bugTitle = `${getTitle(alertSummary)} regression on push ${
-      alertSummary.revision
-    } (${pushDate})`;
+    const bugTitle = `${getFilledBugSummary(
+      alertSummary,
+    )} regression on ${pushDate}`;
```
Comment (ionutgoldan) [inline ui/perfherder/alerts/StatusDropdown.jsx:129]: We could include the "regression on ..." into `getFilledBugSummary()` as well. `alertSummary` param has all the required context.

File: tests/ui/perfherder/helpers_test.jsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      summary_id: 12345,
+      related_summary_id: 12346,
+      manually_created: false,
+      title: 'MySuite Test4',
+    },
+  ],
+  status: 0,
+};
+
+test('getFilledBugSummary', async () => {
```
Comment (ionutgoldan) [inline tests/ui/perfherder/helpers_test.jsx:91]: We should parameterize this test more.

**Review by ionutgoldan — CHANGES_REQUESTED** [review]

File: tests/ui/perfherder/helpers_test.jsx [hunk 7]
```diff
@@ -1,96 +1,62 @@
...
+import testAlertSummaryDownstreamRegressions from './filled_bug_alerts/alert_summary_downstream_regressions';
+import testAlertSummaryRegressionsAndImprovements from './filled_bug_alerts/alert_summary_regressions_and_improvements';
+import testAlertSummaryInvalidRegressions from './filled_bug_alerts/alert_summary_invalid_regressions';
+import testAlertSummaryWithOneRegression from './filled_bug_alerts/alert_summary_with_one_regression';
+import testAlertSummaryRegressionsAndReassigned from './filled_bug_alerts/alert_summary_regressions_and_reassigned';
+import testAlertSummaryRegressionsAndReassignedToOther from './filled_bug_alerts/alert_summary_regressions_and_reassigned_to_other';
 
-test('getFilledBugSummary', async () => {
-  const result = getFilledBugSummary(alertSummaryInput);
+test('getFilledBugSummary downstream regressions', async () => {
```
Comment (ionutgoldan) [inline tests/ui/perfherder/helpers_test.jsx:11]: I just realized there's already a test file for the helpers module. It's named "helpers_test.js". We should move our tests there instead.

**Review by ionutgoldan — CHANGES_REQUESTED** [review]

File: tests/ui/perfherder/filled_bug_alerts/alert_summary_regressions_and_reassigned.json [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+      "manually_created": false,
+      "classifier": null,
+      "starred": false,
+      "classifier_email": null,
+      "backfill_record": null,
+      "title": "Strings PerfUTF8toUTF16AROne macosx1014-64-qr opt "
+    },
+    {
+      "id": 116597,
+      "status": 0,
```
Comment (ionutgoldan) [inline tests/ui/perfherder/filled_bug_alerts/alert_summary_regressions_and_reassigned.json:113]: I think you also need to change the alert' status.

File: tests/ui/perfherder/filled_bug_alerts/alert_summary_regressions_and_reassigned.json [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+      "manually_created": false,
+      "classifier": null,
+      "starred": false,
+      "classifier_email": null,
+      "backfill_record": null,
+      "title": "Strings PerfUTF8toUTF16ARThree macosx1014-64-qr opt "
+    },
+    {
+      "id": 116588,
+      "status": 0,
```
Comment (ionutgoldan) [inline tests/ui/perfherder/filled_bug_alerts/alert_summary_regressions_and_reassigned.json:79]: This status isn't correct.

File: tests/ui/perfherder/filled_bug_alerts/alert_summary_regressions_and_reassigned_to_other.json [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+      "manually_created": false,
+      "classifier": null,
+      "starred": false,
+      "classifier_email": null,
+      "backfill_record": null,
+      "title": "Strings PerfUTF8toUTF16ARThree macosx1014-64-qr opt "
+    },
+    {
+      "id": 116588,
+      "status": 0,
```
Comment (ionutgoldan) [inline tests/ui/perfherder/filled_bug_alerts/alert_summary_regressions_and_reassigned_to_other.json:79]: This isn't the correct status.

File: tests/ui/perfherder/filled_bug_alerts/alert_summary_regressions_and_reassigned_to_other.json [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+      "manually_created": false,
+      "classifier": null,
+      "starred": false,
+      "classifier_email": null,
+      "backfill_record": null,
+      "title": "Strings PerfUTF16toUTF8VIFifteen macosx1014-64-qr opt "
+    },
+    {
+      "id": 116596,
+      "status": 0,
```
Comment (ionutgoldan) [inline tests/ui/perfherder/filled_bug_alerts/alert_summary_regressions_and_reassigned_to_other.json:147]: Neither this one.

**Review by ionutgoldan — CHANGES_REQUESTED** [review]

**Review by ionutgoldan — APPROVED** [review]

**Pull request merged**